### DISCUSSION TITLE
Rebase/transplant a branch with git-town

### DISCUSSION THREAD

**Discussion by sergej-koscejev** [git-town/git-town#3867] [Q&A] [answered]
I thought `git town set-parent` (or a subsequent `git town sync`) would rebase my current branch (and all its descendants) on top of the new parent but instead the new parent is merged into it. Is there a separate Git Town command to do such a rebase/"branch transplant"?

**Comment by kevgo** [comment]
`git sync` can either rebase your branch on top of its parent, or merge the parent into your branch. Which one of the two it does depends on the [sync-feature-strategy setting](https://www.git-town.com/preferences/sync-feature-strategy).

**Reply by sergej-koscejev** [reply]
Let's say I'm on the branch `another-feature` and I have the following config (`gt` is alias for `git-town`):
```
$ gt config

Branches:
  main branch: main
  perennial branches: (none)
  perennial regex: release/.*
  parked branches: (none)
  contribution branches: (none)
  observed branches: (none)

Configuration:
  offline: no
  run pre-push hook: yes
  push new branches: no
  ship deletes the tracking branch: no
  sync-feature strategy: rebase
  sync-perennial strategy: rebase
  sync with upstream: yes
  sync tags: yes

Hosting:
  hosting platform override: (not set)
  GitHub token: (not set)
  GitLab token: (not set)
  Gitea token: (not set)

Branch Lineage:
  release/1.x
    release-feature
      another-feature
```

Now I have decided that I don't want `another-feature` to be on top of `release/1.x` but rather on top of `main`. So I run `gt set-parent` and choose `main`. When I now run `gt sync`, it runs `git rebase main`, bringing not just the commits of `another-feature` but also the commits of `release-feature` on top of `main`. (And if I have `rebase.updateRefs` enabled, then it will also update the `release/1.x` and `release-feature` branches to be based on top of `main`.)

Is there a Git Town command to bring just the commits of `another-feature` on top of main instead? Or do I have to do it manually?

**Reply by kevgo** [reply, marked as answer]
That's a great question! This is planned in #3338. In your scenario you would use the "detach a branch from a stack" feature discussed there. For now you need to do this manually:
- create a new branch
- [cherry-pick](https://git-scm.com/docs/git-cherry-pick) the commits from `another-feature` into it
- delete the old `another-feature` branch

If you [compress](https://www.git-town.com/commands/compress) the `another-feature` branch first, all its changes would be in a single commit, and you have to cherry-pick only that commit.
